templates/dex.yaml

```diff
@@ -0,0 +1,64 @@
+apiVersion: kubeaddons.mesosphere.io/v1alpha1
+kind: Addon
+metadata:
+  name: dex
+  namespace: kubeaddons
+spec:
+  kubernetes:
+    minSupportedVersion: v1.14.0
+  chartReference:
+    chart: stable/dex
+    version: 1.4.0
+    values: |
+      ---
+      image: mhrabovcin/dex
+      imageTag: "d2f240c9c72682913eaff4fd85d119d7d2d0f240"
+      ingress:
+        enabled: true
+        annotations:
+          kubernetes.io/ingress.class: traefik
+          ingress.kubernetes.io/protocol: https
+        path: /dex
+        hosts:
+          - ""
+      ports:
+        - name: https
+          containerPort: 8080
+          protocol: TCP
+      certs:
+        web:
+          create: false
+          secret:
+            tlsName: dex
+      config:
+        issuer: https://dex-kubeaddons.kubeaddons.svc.cluster.local:8080/dex
+        frontend:
+          issuer: Kubernetes
+          theme: mesosphere
+        storage:
+          type: kubernetes
+          config:
+            inCluster: true
+        logger:
+          level: debug
+        web:
+          http: 127.0.0.1:8081
+          https: 0.0.0.0:8080
+          tlsCert: /etc/dex/tls/https/server/tls.crt
+          tlsKey: /etc/dex/tls/https/server/tls.key
+        grpc:
+          addr: 0.0.0.0:5000
+          tlsCert: /etc/dex/tls/grpc/server/tls.crt
+          tlsKey: /etc/dex/tls/grpc/server/tls.key
+          tlsClientCA: /etc/dex/tls/grpc/ca/tls.crt
+        oauth2:
+          skipApprovalScreen: true
+
+        staticClients:
+        # This `id` must by in sync with `dex-k8s-authenticator.yaml` value as well as
+        # kube-apiserver flag `oidc-client-id`.
+        - id: kube-apiserver
+          redirectURIs:
+            - 'https://dex-k8s-authenticator-kubeaddons.kubeaddons.svc.cluster.local:5555/dex-k8s-authenticator/callback/kubernetes-cluster'
```

Comment:
Note that `cluster.local` could vary although we are not setting it at this moment.

Reply:
I'll add the following comment:

```
            # This URI is just a placeholder and it will be replaced during the
            # deployment with URL pointing to the traefik ingress public
            # load balancer.
```